packages/components/src/Autocomplete/Menu.tsx

```diff
@@ -99,15 +101,45 @@ export function Menu({
           Math.min(options.length - 1, highlightedIndex + indexChange),
         );
       }
+      if (menuDiv.current) {
+        scrollMenuIfItemNotInView(menuDiv.current, "down");
+      }
     });
 
     useOnKeyDown("ArrowUp", (event: KeyboardEvent) => {
       const indexChange = arrowKeyPress(event, IndexChange.Previous);
       if (indexChange) {
         setHighlightedIndex(Math.max(0, highlightedIndex + indexChange));
       }
+      if (menuDiv.current) {
+        scrollMenuIfItemNotInView(menuDiv.current, "up");
+      }
     });
 
+    function scrollMenuIfItemNotInView(
+      menuDivElement: HTMLDivElement,
+      direction: "up" | "down",
+    ) {
+      const itemDiv = menuDivElement.querySelector(
+        `button.${classnames(styles.option)}:nth-child(${highlightedIndex +
+          1})`,
+      ) as HTMLButtonElement;
+      if (!itemDiv) {
+        return;
+      }
+      const menuTop = menuDivElement.getBoundingClientRect().top;
+      const itemTop = itemDiv.getBoundingClientRect().top;
+      const itemHeight = itemDiv.getBoundingClientRect().height;
```

Comment:
```suggestion
      const {top: itemTop, height: itemHeight} = itemDiv.getBoundingClientRect();
```

Reply:
I'm so so on this suggestion. It's _slightly_ better as it's not re-calling `itemDiv.getBoundingClientRect` for each variable but it may not be as readable

Reply:
I like this better than getting the bounding rect for each element, so I'm going to use this change 😎